### PR TITLE
Prepare 0.11.0 release

### PR DESCRIPTION
**Pull request by mtreinish**
This commit prepares the repository for the 0.11.0 release. It makes 2
primary changes it moves the release notes for this release to a
separate directory to separate the release notes from future releases.
The second change is a adding a release note for the prelude providing
the highlights for the release.

TODO:

- [x] Wait for other PRs in https://github.com/Qiskit/retworkx/milestone/7
- [x] Revise and edit release notes for 0.11

### CODE REVIEW

**Comment by mtreinish** [comment]
I've marked this as on hold until we're ready to release 0.11. This should be the last commit we merge for 0.11 and once this merges that commit will be the sha1 we use to tag 0.11.0

File: releasenotes/notes/0.11/fix-degree-with-self-loops-5b40c45cc61fcebe.yaml
Comment (jakelishman) [inline releasenotes/notes/0.11/fix-degree-with-self-loops-5b40c45cc61fcebe.yaml:7]:
```suggestion
    Fixed `#517 <https://github.com/Qiskit/retworkx/issues/517>`__.
```

File: releasenotes/notes/0.11/fix-dispatch-3596ef110cc68338.yaml
Comment (jakelishman) [inline releasenotes/notes/0.11/fix-dispatch-3596ef110cc68338.yaml:7]:
```suggestion
    Fixed `#515 <https://github.com/Qiskit/retworkx/issues/515>`__.
```

File: releasenotes/notes/0.11/full-rary-tree-generator-4c533fd73dbd3e48.yaml
Comment (jakelishman) [inline releasenotes/notes/0.11/full-rary-tree-generator-4c533fd73dbd3e48.yaml:5]: If you make the `n` in the maths font, you probably want the `r` as well:
```suggestion
    support for generating a full :math:`r`\ -ary tree of :math:`n` nodes. For example:
```

File: releasenotes/notes/0.11/negative-weights-in-shortest-path-algos-b205b63bd12d1a26.yaml
Comment (jakelishman) [inline releasenotes/notes/0.11/negative-weights-in-shortest-path-algos-b205b63bd12d1a26.yaml:12]:
```suggestion
    Fixed `#525 <https://github.com/Qiskit/retworkx/issues/525>`__.
```

File: releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml
Comment (jakelishman) [inline releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml:8]:
```suggestion
  functions: :func:`~retworkx.bfs_search`, :func:`~retworkx.dfs_search`,
  :func:`~retworkx.dijkstra_search`, and :class:`~retworkx.TopologicalSorter`,
```

File: releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml
Comment (jakelishman) [inline releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml:12]: Do you want to link to `retworkx-core` on crates.io or wherever it's made public?

File: releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml
Comment (jakelishman) [inline releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml:29]: Similar question about linking to crates.io here.

File: releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml
Comment (jakelishman) [inline releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml:33]:
```suggestion
    This new crate only exposes a Rust interface that is general for any
    petgraph graph and can be used by any downstream Rust project that wants
```

File: releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml
Comment (jakelishman) [inline releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml:37]:
```suggestion
    there may be breaking API changes in a subsequent release. While we will
```

File: releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml
Comment (jakelishman) [inline releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml:41]:
```suggestion
    attempt to follow the standard deprecation and stability policy, since
    we're not necessarily fully committed to the current API and without having
    a user outside of retworkx, there may be gaps or issues which require
    breaking changes.
```

**Comment by mtreinish** [comment]
All the release blocking PRs have merged, this is unblocked. Once this PR is reviewed and merges I'll tag that sha1 as the 0.11.0 release

File: releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml
Comment (jakelishman) [inline releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml:48]:
```suggestion
    is set to ``True`` the node attributes from the input NetworkX graph will
    be preserved. The data payload for the output retworkx graph will be
    a dictionary containing the attributes, with an extra ``"__networkx_node__"`` key
    containing the node from NetworkX. For example::
```

(I think - I was struggling to understand this paragraph a little.)

**Review by jakelishman — COMMENTED** [review]
Here are some comments on the diff that I can actually see in the GitHub interface - they're almost entirely minor typos, and a couple of nitpicky changes where I wasn't super clear what was being said.

Looking at the docs I generated locally from this branch, I think there are some typos and things in places that I can't comment on via the diff.  It's a bit late for me to go through the notes manually today, but if you're still waiting on the release in my morning tomorrow, I can do it then and PR to this branch.

File: releasenotes/notes/0.11/barbell-graph-generator-5cc9df7148b1a09e.yaml
Comment (mtreinish) [reply releasenotes/notes/0.11/barbell-graph-generator-5cc9df7148b1a09e.yaml:5]: Yeah that's expected, since it doesn't actually define it's own docstring (it's a rust python module). We just use it as a grouping namespace for the graph generator functions. I'll change this to ``retworkx.generators`` to make it clear (it should render the same way):

```suggestion
    Added a new generator function, :func:`~retworkx.generators.barbell_graph`,
    to the ``retworkx.generators`` module that will generate a barbell graph.
```

File: releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml
Comment (mtreinish) [reply releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml:12]: Yeah that's probably not a bad idea. The link will be: https://crates.io/crates/retworkx-core but there is a bit of a race condition because I won't publish the crate until tagging but since the url is stable it'll resolve once it's there I'll add it now.

File: releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml
Comment (mtreinish) [reply releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml:48]: Yeah, it's a bit weird and hard to explain consisely because it requires knowledge of both networkx and retworkx's APIs. The data payload for:

```
        g = nx.Graph()
        g.add_nodes_from([
          ("A", {"color": "turquoise", "size": "extra large"}),
          ("B", {"color": "fuschia", "size": "tiny"}),
        ])
        rx_graph = rx.networkx_converter(g)
```

is normally just the 2 nodes in `rx_graph` are `"A"` and `"B"`. But if you set `keep_attributes` to `True` the output node payloads get set to a dictionary with any attributes and `__networkx_node__` the networkx node data payload.

File: releasenotes/notes/0.11/add-cartesian-product-f57b4aeb8cbebee7.yaml
Comment (mtreinish) [inline releasenotes/notes/0.11/add-cartesian-product-f57b4aeb8cbebee7.yaml:5]:
```suggestion
    type variants :func:`~retworkx.digraph_cartesian_product` and
```

File: .mailmap
Comment (mtreinish) [reply .mailmap:20]: Yeah, that's probably a good idea. But, I'll leave that for a follow up. It looks like the mailmap in terra for Georgios just changes his real name to use his name instead of his github username.

I only put this here because the git author stats were showing Ivan 4 times and I wanted to unify that before the release .

File: releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml
Comment (mtreinish) [reply releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml]: Done in: https://github.com/Qiskit/retworkx/pull/549/commits/b16f1549d0297a7459cd5332e19b97e316d7b99f

File: releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml
Comment (mtreinish) [reply releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml]: Done in: https://github.com/Qiskit/retworkx/pull/549/commits/b16f1549d0297a7459cd5332e19b97e316d7b99f

**Comment by mtreinish** [comment]
> Here are some comments on the diff that I can actually see in the GitHub interface - they're almost entirely minor typos, and a couple of nitpicky changes where I wasn't super clear what was being said.
> 
> Looking at the docs I generated locally from this branch, I think there are some typos and things in places that I can't comment on via the diff. It's a bit late for me to go through the notes manually today, but if you're still waiting on the release in my morning tomorrow, I can do it then and PR to this branch.

Thanks for the review, I think we'll probably have this merged and release by your morning. But, still feel free to open a PR with any fixes you find after the release too. Since retworkx's docs are self contained we can use reno as intended and any release note fixes will automatically update the docs directly with the sphinx extension so we can get them out in any subsequent release.

File: releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml
Comment (kevinhartman) [inline releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml:26]:
```suggestion
  The 0.11.0 release fully supports Python 3.10. Precompiled binaries
  for Python 3.10 are published on PyPI (previous releases worked with 3.10 but
  required compiling from source to install). This is also the last release to
  support Python 3.6. Starting in retworkx 0.12.0, Python >=3.7 will be required
  to run retworkx. Additionally, for users compiling retworkx from source, this
  will be the last release with a minimum supported Rust version (MSRV) of 1.41.
  In retworkx 0.12.0, the MSRV will be increased to 1.48.
```

File: releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml
Comment (kevinhartman) [inline releasenotes/notes/0.11/prepare-0.11-af688e532712c830.yaml:37]:
```suggestion
    without Python.
```

**Review by kevinhartman — APPROVED** [review]
Just a few minor suggestions.

File: releasenotes/notes/0.11/barbell-graph-generator-5cc9df7148b1a09e.yaml
Comment (mtreinish) [reply releasenotes/notes/0.11/barbell-graph-generator-5cc9df7148b1a09e.yaml:5]: There isn't really a page generated for the module (at least right now). We have a section in the api docs toc tree: https://github.com/Qiskit/retworkx/blob/main/docs/source/api.rst#generators which renders to: https://qiskit.org/documentation/retworkx/dev/api.html#generators but besides that the `retworkx.generators` module doesn't have it's own page in sphinx or even a docstring set right now.

To change this we'd have to add a docstring to the module definition here: https://github.com/Qiskit/retworkx/blob/main/src/generators.rs#L2491 and then add the module to the sphinx doc via `automodule` or `autosummary`. It's probably not a bad idea to do this post-release and we can always backport it as part of a bugfix release.

**Pull request merged**
